classes/openrc.bbclass

```diff
@@ -97,10 +118,46 @@ python openrc_populate_packages() {
         localdata.prependVar("OVERRIDES", f"{pkg}:")
         localdata.setVar(f"OPENRC_SERVICES:{pkg}", " ".join(services))
 
+        # Prepare per-service runlevel variables and escaped service names for shell scripts
```

Comment:
adding variables to the end of the bash script is slick.  If we're going to do that though, we may as well make associative  arrays (see the end of this page, https://mywiki.wooledge.org/BashGuide/Arrays) and use `shlex.quote()` on both the script and runlevel names.

untested:
```python
script_vars="declare -A service_runlevels\n"
for service in services:
    runlevel = # same stuff you have
    script_vars += "service_runlevels[%s]=%s" % (shlex.quote(service), shlex.quote(runlevel))
```

Then in the shell portion
```bash
for script in ${!service_runlevels[@]}; do
    runlevel=${service_runlevels[${script}]}
    # ...
```

Reply:
Arg, yeah you're correct.  I was mistakenly thinking this ran on the build machine which would have a newer bash.

I'm mildly worried about only handling `-` when there's plenty of other characters that would be valid in the in the filename but not in the shell variable name.  This is why I was trying to pack everything into an array.  I suppose we could make a single string with `"service|runlevel service2|runlevel2 ..."` and then iterate through that by splitting on the whitespace first and `|` second.  I'm picking a `|` only because that seems quite unlikely to be used in a filename.

But maybe it's not worth it and we should just throw an error if we get a script/runlevel name outside of `[a-zA-Z_][a-zA-Z_0-9]*`. What do you think?

Reply:
You're right to be concerned about shell variable names, but the real constraint here is BitBake variable naming. Since users define `OPENRC_RUNLEVEL:service-name` in their recipes, we're already limited to BitBake-valid characters. The hyphen is the main problematic character we encounter in practice (e.g., `my-service.initd`), which we handle by converting to underscore.

If we encounter a service name with truly invalid characters, BitBake itself would reject the `OPENRC_RUNLEVEL:invalid@name` variable definition, so we'd get early feedback. I think the current approach is simpler and sufficient.

Reply:
Good point, thanks.